packages/backend-core/src/sql/sql.ts

```diff
@@ -602,24 +605,33 @@ class InternalBuilder {
     } else if (paginate && paginate.limit) {
       foundLimit = paginate.limit
     }
-    // start building the query
-    let query = this.knexWithAlias(knex, endpoint, tableAliases)
-    query = query.limit(foundLimit)
-    if (foundOffset) {
+    // always add the found limit, unless counting
+    if (!counting) {
+      query = query.limit(foundLimit)
+    }
+    // add overall pagination
+    if (!counting && foundOffset) {
       query = query.offset(foundOffset)
     }
+    // add filters to the query (where)
     query = this.addFilters(query, filters, json.meta.table, {
       aliases: tableAliases,
     })
-
     // add sorting to pre-query
-    query = this.addSorting(query, json)
+    if (!counting) {
+      query = this.addSorting(query, json)
+    }
     const alias = tableAliases?.[tableName] || tableName
     let preQuery = knex({
       [alias]: query,
-    } as any).select(selectStatement) as any
+    } as any)
+    if (counting) {
+      preQuery = preQuery.count("* as total")
+    } else {
+      preQuery = preQuery.select(selectStatement)
+    }
     // have to add after as well (this breaks MS-SQL)
-    if (this.client !== SqlClient.MS_SQL) {
+    if (this.client !== SqlClient.MS_SQL && !counting) {
```

Comment:
It just occurred to me, do we not allow sorting for SQL Server? Because before you added the condition at the end here, this was never adding sorting for SQL Server. Is that right?

Reply:
We do allow sorting on SQL Server, essentially if a sort isn't applied it always applies a base sort (I believe).